seismic/xcorqc/utils.py

```diff
@@ -0,0 +1,200 @@
+def split_list(lst, npartitions):
+    k, m = divmod(len(lst), npartitions)
+    return [lst[i * k + min(i, m):(i + 1) * k + min(i + 1, m)] for i in range(npartitions)]
+# end func
+
+def drop_bogus_traces(st, sampling_rate_cutoff=1):
+    """
+    Removes spurious traces with suspect sampling rates.
+    :param st: Obspy Stream
+    :param sampling_rate_cutoff: sampling rate threshold
+    :return: Input stream is updated inplace
+    """
+    badTraces = [tr for tr in st if tr.stats.sampling_rate < sampling_rate_cutoff]
+
+    for tr in badTraces: st.remove(tr)
+# end func
+
+def _get_stream_00T(fds, net, sta, cha, start_time, end_time,
+                      baz=None, automerge=False, trace_count_threshold=200,
+                      logger=None, verbose=1):
+
+    stations = fds.get_stations(start_time, end_time, network=net, station=sta)
+
+    stations_nch = [s for s in stations if 'N' == s[3][-1].upper() or '1' == s[3][-1]]  # only N channels
+    stations_ech = [s for s in stations if 'E' == s[3][-1].upper() or '2' == s[3][-1]]  # only E channels
+
+    stt = None
+    if (len(stations_nch) > 0 and len(stations_nch) == len(stations_ech)):
+        for codesn, codese in zip(stations_nch, stations_ech):
+
+            stn = fds.get_waveforms(codesn[0], codesn[1], codesn[2], codesn[3],
+                                    start_time,
+                                    end_time,
+                                    automerge=automerge,
+                                    trace_count_threshold=trace_count_threshold)
+            ste = fds.get_waveforms(codese[0], codese[1], codese[2], codese[3],
+                                    start_time,
+                                    end_time,
+                                    automerge=automerge,
+                                    trace_count_threshold=trace_count_threshold)
+
+            if (len(stn) == 0): continue
+            if (len(ste) == 0): continue
+            
+            drop_bogus_traces(stn)
+            drop_bogus_traces(ste)
+
+            # Merge station data. Note that we don't want to fill gaps; the
+            # default merge() operation creates masked numpy arrays, which we can use
+            # to detect and ignore windows that have gaps in their data.
+            try:
+                stn.merge()
+                ste.merge()
+
+                max_start_time = np.max([stn[0].stats.starttime, ste[0].stats.starttime])
+                min_end_time   = np.min([stn[0].stats.endtime, ste[0].stats.endtime])
+
+                stn = stn.slice(starttime = max_start_time, endtime= min_end_time)
+                ste = ste.slice(starttime=max_start_time, endtime=min_end_time)
+            except Exception as e:
+                if logger: logger.warning('\tFailed to merge traces..')
+                st = None
+                raise
+            # end try
+
+            bazr = np.radians(baz)
+            tdata = - ste[0].data * np.cos(bazr) + stn[0].data * np.sin(bazr)
```

Comment:
Consider removing space after unary minus sign, as is customary in most modern style guides. As it is I think it is quite hard to see.

Reply:
True; done.